### PR TITLE
FreeBSD node definition and instructions

### PR DESCRIPTION
**Pull request by thepurplebuffalo**
Cleaned the commit history for this PR.

### CODE REVIEW

File: node-definitions/freebsd/FreeBSD.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  label: FreeBSD
+sim:
+  linux_native:
+    libvirt_domain_driver: kvm
+    driver: server
+    disk_driver: sata
+    ram: 1024
+    cpus: 1
+    cpu_limit: 100
+    nic_driver: e1000
```
Comment (xorrkaz) [inline node-definitions/freebsd/FreeBSD.yaml:27]: FreeBSD can make use of VirtIO in the default kernel.  I think it performs a bit better.

File: node-definitions/freebsd/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Terraform to provision the CML node.
+
+## Nuage-init: MAC address required
+
+If you are attempting to use Nuage-init in order to set an IP address please
+have a look at the [source code](https://github.com/freebsd/freebsd-src/blob/f35ccf46c7c6cb7d26994ec5dc5926ea53be0116/libexec/nuageinit/nuageinit#L283-L293).
+
+Nuage-init will explicitly check to see that there is an ```ethernets```
+section with a ```match``` subsection.  The only working match clause is the
+```macaddress``` clause.  If you do not explicitly state the MAC address of
```
Comment (xorrkaz) [inline node-definitions/freebsd/README.md:42]: Not a knock on you, but this is **lame**.  The good news is in CML 2.8 you'll be able to customize MAC addresses.

File: node-definitions/freebsd/FreeBSD.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+id: FreeBSD
+general:
+  nature: server
+  description: FreeBSD with "nuage-init" (a limited version of cloud-init).
+  read_only: false
+device:
+  interfaces:
+    has_loopback_zero: true
+    physical:
+      - em0
```
Comment (xorrkaz) [inline node-definitions/freebsd/FreeBSD.yaml:10]: Why not add multiple NICs like you did with OpenBSD?  I think at least two would be useful.

**Review by xorrkaz — CHANGES_REQUESTED** [review]
Thanks for doing this!

File: node-definitions/freebsd/README.md (hunk 2 above)
Comment (thepurplebuffalo) [reply node-definitions/freebsd/README.md:42]: Agreed.  I saw the news about CML 2.8 about 20 minutes after I submitted the PR.  It gave me hope.

File: node-definitions/freebsd/FreeBSD.yaml (hunk 3 above)
Comment (thepurplebuffalo) [reply node-definitions/freebsd/FreeBSD.yaml]: Updated!

**Pull request merged**